src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -23,7 +23,10 @@ public static bool IsBlittableType(TypeDesc type)
                 && !baseType.IsWellKnownType(WellKnownType.Object)
                 && !baseType.IsWellKnownType(WellKnownType.ValueType);
 
-            if (hasNonTrivialParent && !IsBlittableType(baseType))
+            // Type is blittable only if parent is also blittable and is not empty.
+            if (hasNonTrivialParent && !IsBlittableType(baseType) ||
+                (!type.IsValueType && baseType is MetadataType baseMdType && !baseType.IsWellKnownType(WellKnownType.Object) &&
```

Comment:
Why `!type.IsValueType`?

Reply:
Ah IsValueType is too strict, we just need `hasNonTrivialParent` to skip those well-known bases types from this check.